---
title: "GENOMIC SELECTION MODELS - BLASSO, rrBLUP, BWGS"
author: "Ajay Kumar, International Rice Research Institute (IRRI, South Asia Hub)"
date: "12/10/2021"
output:
  pdf_document: default
  html_document: default
---
## DeepMap
Deep Learning has rapidly evolved and is now routinely being used in prediction-based studies in crops improvement. In this study, an effort was made to develop the DeepMap, a unique deep learning enabled python package for genotype to phenotype prediction. It can be easily extended and used in other crops such as Maize, Wheat, Soybean and even Cattle, Loblolly pine and Humans. It uses epistatic interactions for data augmentation and outperforms existing state-of-the-art machine/deep learning models such as Bayesian LASSO, GBLUP, DeepGS, and dualCNN. It is hosted on Python Package Index for ease-of-use to encourage reproducibility with four-line code execution which import libraries, takes genotypic and phenotypic data, invoking model for training/testing data and storing results. The DeepMap can be further improve by adding environmental interactions, incorporating nascent architecture of deep learning such as GANs, autoencoders, transformers etc., and development of graphical user interface would increase the user community with ease-of-use.

This R-Markdown document is useful for genotype to phenotype prediction using RR-BLUP, G-BLUP,EN-Model, BRNN-Model and BL_BLUP.

#### 1.1 Importing Library
```{r}
library(BWGS)
```

#### 1.2 Set working directory
```{r b}
setwd("D:/_IRRI-SOUTH ASIA/Paper-1_DeepMap/Paper_writing/R 11/Github_DeepMap/GS_Models - BLASSO, rrBLUP, BWGS, Python based DeepGS") #Please set your working directory here
```


#### 1.4 Give genotypic and phenotypic input files
```{r}
geno = read.csv("SNP.csv") #Replace it with your genotypic file name, please refer the format (refer SNP.csv file) before running the code
row.names(geno) <- geno[,1] #Change your row-names with designation id here, needed for prediction
pheno = read.csv("pheno.csv") #Replace it with your phenotypic file name, please refer the format (refer pheno.csv file) before running
pheno = as.vector(pheno[,2])
names(pheno) <- geno[,1] #Changing row-names with designation id here, needed for prediction
geno = geno[,2:ncol(geno)]
```

#### 1.5 Choose model to train and test your data
```{r}
#Model - 1

GBLUP_results = bwgs.cv(geno,pheno,geno.impute.method = "mni" ,predict.method = "GBLUP",nFolds = 10,nTimes = 1) #You may change the parameters as per your experiment analysis
GBLUP_results=as.data.frame(gblup_results$summary) #Saving results in a variable
```

```{r}
#Model - 2
EN_results = bwgs.cv(geno,pheno,geno.impute.method = "mni" ,predict.method = "EN",nFolds = 10,nTimes = 1)
EN_results=as.data.frame(EN_results$summary)
```

```{r}
#Model - 3
BL_results = bwgs.cv(geno,pheno,geno.impute.method = "mni" ,predict.method = "BL",nFolds = 10,nTimes = 1)
BL_results=as.data.frame(BL_results$summary)
```
```{r}
#Model - 4
RR_results = bwgs.cv(geno,pheno,geno.impute.method = "mni" ,predict.method = "RR",nFolds = 10,nTimes = 1)
RR_results=as.data.frame(RR_results$summary)
```
  
```{r}
#Model - 5
BRNN_results = bwgs.cv(geno,pheno,geno.impute.method = "mni" ,predict.method = "BRNN",nFolds = 10,nTimes = 1)
BRNN_results=as.data.frame(BRNN_results$summary)

```
#### 1.5 Saving the results
```{r}
#Combining the model outputs; Uncomment below to save the results.
#results=cbind(GBLUP_results,EN_results,BL_results,RR_results,BRNN_results)
#write.csv(results,file = paste0("results/results_",i))
```
